src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx

```diff
@@ -111,50 +108,41 @@ const ProjectEditContainer: React.FC<ProjectEditContainerProps> = ({ project, ex
     remove: removeConstraint
   } = useFieldArray({ control, name: 'constraints' });
   const { mutateAsync } = useEditSingleProject(project.wbsNum);
+  const [projectManagerId, setprojectManagerId] = useState<string | undefined>(project.projectManager?.userId.toString());
+  const [projectLeadId, setProjectLeadId] = useState<string | undefined>(project.projectLead?.userId.toString());
 
   if (allUsers.isLoading || !allUsers.data) return <LoadingIndicator />;
   if (allUsers.isError) {
     return <ErrorPage message={allUsers.error?.message} />;
   }
 
   const users = allUsers.data.filter((u) => u.role !== 'GUEST');
+  const onSubmit = async (data: any) => {
+    const { name, budget, summary, bomLink, googleDriveFolderLink, taskListLink, slideDeckLink } = data;
+    const rules = data.rules.map((rule: any) => rule.rule || rule);
 
-  const onSubmit = async (data: ProjectEditFormInput) => {
-    const {
-      name,
-      budget,
-      summary,
-      bomLink = '',
-      googleDriveFolderLink = '',
-      taskListLink = '',
-      slideDeckLink = '',
-      projectLeadId = 0,
-      projectManagerId = 0
-    } = data;
-    const rules = data.rules.map((rule) => rule.rule);
     const goals = mapBulletsToPayload(data.goals);
     const features = mapBulletsToPayload(data.features);
     const otherConstraints = mapBulletsToPayload(data.constraints);
 
-    const payload = {
-      name,
-      budget,
-      summary,
-      bomLink,
-      googleDriveFolderLink,
-      taskListLink,
-      slideDeckLink,
-      projectId: project.id,
-      crId: parseInt(data.crId),
-      rules,
-      goals,
-      features,
-      otherConstraints,
-      projectLeadId,
-      projectManagerId
-    };
-
     try {
+      const payload = {
+        name,
+        budget: parseInt(budget),
```

Comment:
why did you change this line?

Reply:
I saw what we did with the crId and thought it would be a good idea